awesome/src/common/object.rs

```diff
@@ -196,7 +196,7 @@ impl<'lua, S: ObjectStateType> ToLua<'lua> for Object<'lua, S> {
 impl<'lua, S: ObjectStateType> FromLua<'lua> for Object<'lua, S> {
     fn from_lua(val: Value<'lua>, _lua: &'lua Lua) -> rlua::Result<Self> {
         if let Value::UserData(obj) = val {
-            Object::cast(obj)
+            Ok(obj.into())
```

Comment:
This is the type check I removed in order to get this to work.

if it's left as is this crashes when loading the configuration while setting the index miss handler in `gear/objects/properties.lua` on line 83 or so. When I dug into it a bit, it expects it to be a screen state (i.e. `S` = `ScreenState`) but it's not that. However when using the `into` here it just works oddly enough?